### DISCUSSION TITLE
A tool for searching all the pdfs Ina directory

### DISCUSSION THREAD

**Discussion by dummifiedme** [pymupdf/PyMuPDF#833] [Ideas]
I like how google drive searches through the contents of a document and provides with suitable docs as per the search. I was wondering if there could be a tool or a possible implementation using PyMuPDF for at least a collection of PDFs?

A simple search, that provides a list of possible matches with or without a bit of context. Searches even your notes or annotations. 

If it is possible, what could be the requirements —  as in the libraries or algorithms or anything — likes pre-requisite to start on sucha project?

**Comment by JorjMcKie** [comment]
Have example output (screen print)?
Should be easy-peasy to do.

**Reply by dummifiedme** [reply]
Will it search for exactly the phrases or could it do something like a fuzzy logic (is it?)?

**Reply by JorjMcKie** [reply]
Depends on how it will be programmed. An idea may be to indeed use regular expressions this time, extract all pages' text as one string and then apply the `re` criterion. if we have a hit for a page, we append it to our list.

As I wrote, I would like to see an example for that Google selection, I am not using it myself.

**Reply by dummifiedme** [reply]
I would attach the screenshot tomorrow morning.
I basically mean, if I type `this lie`, it should show results like 'this is a lie', 'this line', 'this is absolutely a lie'... Basically all the approximated matches as well.

I found a library that does something like that called [FuzzyWuzzy](https://www.geeksforgeeks.org/fuzzywuzzy-python-library/). But yes, you are right. We can use regex as well.
